### PR TITLE
fix: updated redirect data urls

### PR DESCRIPTION
**Pull request by taty2010**
### Summary

This update will normalize data redirects 

https://github.com/netlify/next-runtime/issues/1800

### Test plan

1. Visit the Deploy Preview ([insert link to specific page]()) ...

### Standard checks:

- [ ] Check the Deploy Preview's Demo site for your PR's functionality
- [ ] Add docs when necessary

---

🧪 Once merged, make sure to update the version if needed and that it was published correctly.

### CODE REVIEW

File: packages/runtime/src/templates/edge-shared/utils.ts [hunk 1]
```diff
@@ -6,7 +6,24 @@ export interface FetchEventResult {
   waitUntil: Promise<any>
 }
 
-type NextDataTransform = <T>(data: T) => T
+function normalizeDataUrl(redirect, url) {
```
Comment (nickytonline) [inline packages/runtime/src/templates/edge-shared/utils.ts:9]: I forgot to mention it on our Zoom, but it'd be good to have a link to where this is done in Next.js code to give more context.

Comment (taty2010) [reply packages/runtime/src/templates/edge-shared/utils.ts:9]: That's a great idea, thanks!

**Comment by taty2010** [comment]
The middleware-redirect test is failing within the PR but not locally.

File: packages/runtime/src/templates/edge-shared/utils.ts [hunk 2]
```diff
@@ -250,6 +266,12 @@ export const buildResponse = async ({
     res.headers.set('x-nextjs-redirect', relativizeURL(redirect, request.url))
   }
 
+  const nextRedirect = res.headers.get('x-nextjs-redirect')
```
Comment (nickytonline) [inline packages/runtime/src/templates/edge-shared/utils.ts:269]: We tested this together and it works well. I still have no idea why we have to set the header first, get it and the reset it. If anyone knows why, I'm curious. 🕵🏻

**Review by nickytonline — APPROVED** [review]
I see it's all good now. Ubuntu tests are still hanging, but given the other OSes for tests are passing, going to go ahead and approve. If the Ubuntu tests are still hanging, try restarting the job.

:shipit:

**Comment by nickytonline** [comment]
Just checking in here @taty2010. Anything else need to be done here or are the test failures valid failures?

**Review by sarahetter — APPROVED** [review]

**Pull request merged**